repos/system_upgrade/common/actors/peseventsscanner/libraries/pes_event_parsing.py

```diff
@@ -0,0 +1,252 @@
+import json
+import os
+from collections import namedtuple
+from enum import IntEnum
+from itertools import chain
+
+from leapp import reporting
+from leapp.exceptions import StopActorExecution
+from leapp.libraries.common import fetch
+from leapp.libraries.common.config import architecture
+from leapp.libraries.stdlib import api
+
+_Package = namedtuple('Package', ['name',         # str
+                                  'repository',   # str
+                                  'modulestream'  # (str, str) or None - a module stream
+                                  ])
+
+
+class Package(_Package):  # noqa: E0102; pylint: disable=function-redefined
+    def __repr__(self):
+        ms = ''
+        if self.modulestream:
+            ms = '@{0}:{1}'.format(*self.modulestream)
+        return '{n}:{r}{ms}'.format(n=self.name, r=self.repository, ms=ms)
+
+    def __hash__(self):
+        return hash((self.name, self.modulestream))
+
+    def __eq__(self, other):
+        return (self.name, self.modulestream) == (other.name, other.modulestream)
+
+
+Event = namedtuple('Event', ['id',            # int
+                             'action',        # An instance of Action
+                             'in_pkgs',       # A set of Package named tuples
+                             'out_pkgs',      # A set of Package named tuples
+                             'from_release',  # A tuple representing a release in format (major, minor)
+                             'to_release',    # A tuple representing a release in format (major, minor)
+                             'architectures'  # A list of strings representing architectures
+                             ])
+
+
+class Action(IntEnum):
+    PRESENT = 0
+    REMOVED = 1
+    DEPRECATED = 2
+    REPLACED = 3
+    SPLIT = 4
+    MERGED = 5
+    MOVED = 6
+    RENAMED = 7
+
+
+def get_pes_events(pes_json_directory, pes_json_filename):
+    """
+    Get all the events from the source JSON file exported from PES.
+
+    :return: List of Event tuples, where each event contains event type and input/output pkgs
+    """
+    try:
+        all_events = parse_pes_events(fetch.read_or_fetch(pes_json_filename, directory=pes_json_directory,
+                                                          allow_empty=True))
+        arch = api.current_actor().configuration.architecture
+        events_matching_arch = [e for e in all_events if not e.architectures or arch in e.architectures]
+        return events_matching_arch
+    except (ValueError, KeyError):
+        title = 'Missing/Invalid PES data file ({}/{})'.format(pes_json_directory, pes_json_filename)
+        summary = 'Read documentation at: https://access.redhat.com/articles/3664871 for more information ' \
+            'about how to retrieve the files'
```

Comment:
I personally prefer this kind of formatting over backslash continuation
```suggestion
        summary = ('Read documentation at: https://access.redhat.com/articles/3664871 for more information '
                   'about how to retrieve the files')
```

Not necessary, but It feels more readable to me

Reply:
Resolved.